src/plots/plots.js

```diff
@@ -468,7 +477,29 @@ plots.supplyDefaults = function(gd) {
 
     // first fill in what we can of layout without looking at data
     // because fullData needs a few things from layout
-    plots.supplyLayoutGlobalDefaults(newLayout, newFullLayout);
+
+    if(oldFullLayout._initialAutoSizeIsDone) {
+        // coerce the updated layout while preserving width and height
+        var oldWidth = oldFullLayout.width,
```

Comment:
Why do we need this?

Doesn't `plotAutoSize` [copy](https://github.com/plotly/plotly.js/blob/19a37357341e7e098ce2c20ee98f65965c2c5d66/src/plot_api/plot_api.js#L2558-L2559) the calculated `width` and `height to `layout.width` and `layout.height` respectively?